azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -270,7 +271,7 @@ func resourceArmVirtualMachineScaleSet() *schema.Resource {
 			},
 
 			"os_profile_linux_config": {
-				Type:     schema.TypeSet,
+				Type:     schema.TypeList,
```

Comment:
this could be a bigger scope than we anticipated, given how the ARM API's return the latest snapshot of a resource; I think it'd be worth reverting this back to a Set for the moment (with a separate test function, and the hash function; and then spending some time on both VM/VMSS for 2.0?)